frontend/components/player/playing_controls.jsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { GrCirclePlay } from "react-icons/gr";
+import { BiSkipPrevious, BiSkipNext, BiShuffle } from "react-icons/bi";
+import { BsRepeat, BsRepeat1, BsPauseCircle } from "react-icons/bs";
+import { useEffect } from "react";
+
+const PlayingControls = ({
+	isPlaying,
+	trackProgress,
+	togglePlay,
+	toPrevTrack,
+	toNextTrack,
+	toggleShuffle,
+}) => {
+	useEffect(() => {}, [isPlaying]);
+	return (
+		<div className="playing-controls">
+			<BiShuffle
+				className="player__grey-icon repeat-shuffle-icon"
+				aria-label="Shuffle"
+				onClick={toggleShuffle}
+			/>
+			<BiSkipPrevious
+				className="player__grey-icon"
+				aria-label="Previous"
+				onClick={toPrevTrack}
+			/>
+			{isPlaying ? (
+				<BsPauseCircle
+					className="player__white-icon"
+					aria-label="Pause"
+					onClick={togglePlay}
+				/>
+			) : (
+				<GrCirclePlay
+					className="player__white-icon"
+					aria-label="Play"
+					onClick={togglePlay}
+				/>
+			)}
+			<BiSkipNext
+				className="player__grey-icon"
+				aria-label="Next"
+				onClick={toNextTrack}
+			/>
+			<BsRepeat
+				className="player__grey-icon repeat-shuffle-icon"
+				aria-label="Repeat"
```

Comment:
not implemented, can you add a `// TODO: implement` or raise a `new Error("Not implemented")` in the click handler, this way it's very obvious what's happening

Reply:
Ty for this feedback; wasn't aware I could do that.